File: src/main/java/com/fish/service/impl/AcceptedServiceImpl.java
package com.fish.service.impl;

import com.fish.entity.pojo.Accepted;
import com.fish.mapper.AcceptedMapper;
import com.fish.service.AcceptedService;
import com.mybatisflex.spring.service.impl.ServiceImpl;
import org.springframework.stereotype.Service;

import java.util.ArrayList;

@Service
public class AcceptedServiceImpl extends ServiceImpl<AcceptedMapper, Accepted> implements AcceptedService {
    @Override
    public int addAccepted(Accepted accepted) {
        return mapper.insert(accepted);
    }
    @Override
    public ArrayList<Accepted> getAccepts() {
        return (ArrayList<Accepted>) mapper.selectAll();
    }
    @Override
    public ArrayList<Accepted> getMyAccepts(String userId) {
        return mapper.getMyAccepts(userId);
    }
    @Override
    public ArrayList<Accepted> getTopicAccepts(Long topicId) {
        return mapper.getTopicAccepts(topicId);
    }
}

File: src/main/kotlin/com/fish/keXieOpenJudge/service/accepted/impl/AcceptedServiceImpl.kt
package com.fish.keXieOpenJudge.service.accepted.impl

import com.fish.keXieOpenJudge.entity.pojo.assessment.Accepted
import com.fish.keXieOpenJudge.mapper.assessment.AcceptedMapper
import com.fish.keXieOpenJudge.service.accepted.AcceptedService
import com.mybatisflex.spring.service.impl.ServiceImpl
import org.springframework.stereotype.Service

@Service
class AcceptedServiceImpl : ServiceImpl<AcceptedMapper, Accepted>(), AcceptedService {
    override fun addAccepted(accepted: Accepted): Int {
        return mapper!!.insert(accepted)
    }

    override fun getAccepts(): ArrayList<Accepted> {
        return mapper!!.selectAll() as ArrayList<Accepted>
    }

    override fun getMyAccepts(userId: String): ArrayList<Accepted> {
        return mapper!!.getMyAccepts(userId)
    }

    override fun getTopicAccepts(topicId: Long): ArrayList<Accepted> {
        return mapper!!.getTopicAccepts(topicId)
    }
}
